lightly/models/examples/moco_system.py

```diff
@@ -0,0 +1,67 @@
+import pytorch_lightning as pl
+import torch
+
+from lightly.models.modules.heads import MoCoProjectionHead
+
+import lightly
+from lightly.models.modules.momentum_encoder import MomentumEncoder
+
+
+class MocoSystem(pl.LightningModule):
+    def __init__(self, dataloader_kNN, num_classes):
+        super().__init__(dataloader_kNN, num_classes)
+        # create a ResNet backbone and remove the classification head
+        resnet = lightly.models.ResNetGenerator('resnet-18', num_splits=8)
```

Comment:
I hate that the parameters are hard-coded here and below. IMO this is not an option except if we move the `examples` directory to the top-level (same level as `docs` and `tests`).

Alternatives:
- Pass them as arguments
- Pass `config` objects as arguments to avoid too many arguments
    > "If you have a procedure with 10 parameters, you probably missed some." ~ [Perlisims](http://www.cs.yale.edu/homes/perlis-alan/quotes.html)
- Pass projection head, backbone, loss, and optimizer as arguments to the constructor (they have their own default parameters which we could use)
- We brainstorm and try to find another solution

Reply:
If we have some arguments or even a config object, we have the following disadvantages:
- more maintenance and documentation work
- It is less transparent which parameters are used where and why. E.g. will there be one config object for all models or one for each model? Can I pass a config object with `momentum=0.999` also to the SimCLR Model?
- We can hardly cover all parameters. E.g. what about the scheduler and its parameters? What about using or not using a symmetric loss?
- We become similar to pytorch lightning bolt. 

I agree that we can easily put the examples to the top-level or as an own subpart of the documentation. I like the idea of brainstorming a little more on that.

Reply:
One general problem might be that we don't have a clear target group of the pip package yet. This is also covered in another issue: https://github.com/lightly-ai/lightly/issues/479

Perhaps we should have a discussion about who we want to target with our pip package.